### ISSUE TITLE
TypeError: Cannot read property 'split' of undefined in Ubuntu

### ISSUE THREAD

**Issue by Sharcoux** [saadtazi/firefox-profile-js#136]
The method `readProfiles` from the file `profile_finder.js` seems to be looking for `profiles.ini` to the wrong place in Ubuntu.

The file is located at `/home/<me>/snap/firefox/common/.mozilla/firefox/` in my computer, but `this.directory` is undefined during the execution, and, anyway, the `locateUserDirectory` would return `/home/<me>/.mozilla/firefox`.

**Comment by Sharcoux** [comment]
I'm actually using the `web-ext` command line module from Mozilla that relies on that project. They seem to be using the ProfileFinder class. Besides that, they do things I'm not fully understanding...

**Comment by saadtazi** [comment]
Is there a flag that allows to provide profile directory using the cli? According to the [doc](https://extensionworkshop.com/documentation/develop/web-ext-command-reference/), the flag `-p` (`--firefox-profile`) allows you to specify an absolute path or a profile name. Did you try that flag with an absolute path? something like `npx web-ext -p /home/<me>/snap/firefox/common/.mozilla/firefox/your-profile-dir ...`

**Comment by Sharcoux** [comment]
Yes. But anyway, after I fix the path, I end up with: [this issue](https://github.com/mozilla/web-ext/issues/1696) which is not related to your lib.

The only thing that might be done in this lib is to update `locateUserDirectory` so that it also check in `~/snap/firefox/common/.mozilla/firefox/` if it was not found in `${HOME}/.mozilla/firefox`

**Comment by saadtazi** [comment]
Happy that your (first) issue is solved.

Good idea for `snap` directory. I am accepting PRs if you're interested in contributing 😃

**Comment by Rob--W** [comment]
Copy-pasting from https://github.com/mozilla/web-ext/issues/2826#issuecomment-1739150954:

The error is in firefox-profile: https://github.com/saadtazi/firefox-profile-js/blob/932b40ea8758f22b938af653709767e50a26b98d/lib/profile_finder.js#L55-L57

When reading the file fails (e.g. due to it not existing), supposedly a callback should be called with the error. At line 56, the logic should not only invoke the callback with the error, but also return. Not returning results in the next line trying to parse non-existing content, which results in [the error seen above](https://github.com/mozilla/web-ext/issues/2826#issuecomment-1663758241).

**Comment by saadtazi** [comment]
@Rob--W : you are right, I should fix that. I'll give it a try tonight.

**Comment by saadtazi** [comment]
Of course, please do, no problem! If that makes you feel better, I wrote that package approximately one month after starting nodejs ☺️

**Comment by lissyx** [comment]
@saadtazi this should be closed right?

**Comment by saadtazi** [comment]
True! Thank you
